pages/docs/why-effect.mdx

```diff
@@ -0,0 +1,70 @@
+
+In "typical" TypeScript, without Effect, we typically write code that assumes that a function is either successful or throws an exception. Take this trivial example of division:
+
+```ts twoslash
+function divide(a: number, b: number): number {
+  if (b === 0) {
+    throw new Error("Cannot divide by zero");
+  }
+  return a / b;
+}
+```
+Based on the types, we have no idea that this function can throw an exception. We can only find out by reading the code. This may not seem like much of a problem when you only have one function in your codebase, but when you have hundreds or thousands, it really starts to add up. It's easy to forget that a function can throw an exception, and it's easy to forget to handle that exception.
+
+Often, we will do the "easiest" thing and just wrap the function in a `try/catch` block. This is a good first step to prevent your program from crashing, but it doesn't make it any easier to manage or understand our complex application/library. We can do better.
+
+## The Effect Pattern
+
+While Effect is a vast ecosystem of many different tools, if it had to be reduced down to just one unique idea, it would be the following:
+
+Effect's major unique insight is that we can use the type system to track _errors_ and _"dependencies"_ (more on this later), not only _success_ values as shown in the divide example above.
+
+Here's the same divide function from above, but with the Effect pattern:
+
+```ts twoslash
+import * as Effect from "@effect/io/Effect"
+
+function divide(a: number, b: number): Effect.Effect<never, Error, number> {
+  if (b === 0) {
+    return Effect.fail(new Error("Cannot divide by zero"));
+  }
+  return Effect.succeed(a / b);
+}
+```
+
+Notice how looking at the type signature tells you exactly what dependencies the function needs (`never`, as in, no dependencies), what error(s) it can throw (`Error`), and what success value(s) it returns (`number`). This function no longer throws an exception, and you can cleanly pass on the error to the caller of this function.
+
+Errors now become values, just like your success values. Effect gives us many functions to make managing errors and success values ergonomic.
+
+Additionally, tracking dependencies in your types allows you to write modules separate from your implementation detail. For example, your core domain logic (e.g. creating a user account) can depend on some external API (e.g. sending an account invite email), and you can write that API as an interface/service (e.g. email service). When you write tests, you can easily swap out the API for a mock implementation (e.g. mock email service) and test your core logic without having to worry about the implementation details of the API, and without changing any of your core logic. Don't worry if this doesn't make sense yet. It's one of Effect's most powerful concepts that you will learn more about as you go through the [Core Concepts](../core-concepts/effect).
```

Comment:
This is my least favorite section - I didn't immediately have an intuition for how to describe this better. 

This is one of my favorite Effect features but I only started understanding it after using it so I have a hard time describing it generally without an example.

Reply:
I wouldn't approach it from the perspective of tracking dependencies, this specific framing in a general sense is called "effect tracking" and, while loved by many, it is considered "useless" for business by many others (including myself). What is kind of generally agreed as being valuable is having a computation which is context dependent, for example the reasons you'd use a context in react is to provide external information to a component in the same way the effect context is used to provide information to a program - the provided information can be dependencies / flags / configs / caches / and really anything

Reply:
Thanks for this feedback! I have a better sense of how to frame it now. Will revisit